pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -610,6 +627,19 @@ class Utils {
 
         BulkChange bc = new BulkChange(j)
         try {
+            // Check if QuietPeriod option is specified
+            QuietPeriod quietPeriod = (QuietPeriod) rawOptions.find { it instanceof QuietPeriod }
+            if (quietPeriod != null) {
+                j.setQuietPeriod(quietPeriod.quietPeriod)
+            } else {
+                String quietPeriodName = Jenkins.getActiveInstance().getDescriptorByType(QuietPeriod.DescriptorImpl.class)?.getName()
+                // If the quiet period was set by the previous build, wipe it out.
+                System.err.println("options: ${previousOptions}")
+                if (quietPeriodName != null && previousOptions.contains(quietPeriodName)) {
+                    j.setQuietPeriod(0)
```

Comment:
Unless declarative previously set the quiet period to 0 in some other way, I think you should use `Jenkins.$GETTER.getQuietPeriod()` here to use the default quiet period in Jenkins (normally 5) instead. (WorkflowJob#getQuietPeriod does that as well if quietPeriod is null)

Reply:
Good catch!